src/main/java/Deadline.java

```diff
@@ -0,0 +1,52 @@
+package main.java;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a deadline task, which is a task that need to be done before a specific date/time.
+ */
+public class Deadline extends Task {
+
+    private LocalDate byDate;
+
+    /**
+     * Constructs a Deadline object with the specified description and deadline time.
+     *
+     * @param description The description of the deadline task.
+     * @param by          The due date/time for the deadline task.
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.byDate = LocalDate.parse(by);
+    }
+    /**
+     * Constructs a Deadline object with the specified description and deadline time.
+     *
+     * @param description The description of the deadline task.
+     * @param by          The due date/time for the deadline task.
+     * @param bool        The completion status for the deadline task.
+     */
+    public Deadline(String description, String by, Boolean bool) {
```

Comment:
Ambiguous name. Perhaps a more boolean-sounding name like isDone could be used instead?

Reply:
yepp good catch, I will change that